### ISSUE TITLE
Locale property in dictionaries.xcu should be in the form of "en-US" instead of "eng"

### ISSUE THREAD

**Issue by n8marti** [silnrsi/oxttools#14]
**LibreOffice recognizes existing languages by the 5-character language tag (and possibly also with the shorter 2-letter version) rather than the 3-letter language ISO code.**
I used makeoxt to create a spell check dictionary for Sango (sag), and it failed to be properly recognized by LibreOffice. I was perplexed because I had already done the same for Kaba (ksp), and it worked without issue. After trying a different ISO code in the makeoxt command, I was able to successfully create the dictionary and have it recognized. However, it still didn't work properly because Sango, a LWC in the Central African Republic, was already known to LO, and if I set the paragraph to the known "Sango" language, my spell check dictionary didn't get activated. It was only after I looked at the Spanish, Venezuela (es-VE) dictionary that I saw that it used "es-VE" as it's Locale name in dictionaries.xcu, whereas makeoxt uses the 3-letter ISO code.

This appears to be an oversight due to makeoxt generally being used with less-common languages that aren't already known to LibreOffice, so the generated locale name isn't normally an issue.

Here are the two versions, the "original" as generated by makeoxt, and the working version after changing the Language property:
- [dict-sango.orig.oxt.zip](https://github.com/silnrsi/oxttools/files/6108364/dict-sango.orig.oxt.zip)
- [dict-sango.oxt.zip](https://github.com/silnrsi/oxttools/files/6108365/dict-sango.oxt.zip)

**Comment by DavidLRowe** [comment]
The BCP 47 standard specifies that one use a two-letter code (ISO 639-1) to specify the language if one exists, otherwise use the three-letter code (ISO 639-3). The makeoxt script has a provision for changing "sg" to "sg-CF" to match LibreOffice's expectations, but since you specified "sag" rather than "sg", that check was bypassed.

Perhaps makeoxt should also check that an ISO 639-3 code is not used if an ISO 639-1 code exists?

**Comment by DavidLRowe** [comment]
@mhosken Would it be feasible to add a check to see that the language tag uses an ISO 639-1 code if one exists rather than an ISO 639-3 code?

**Comment by n8marti** [comment]
I think that would be helpful. I was ignorant of the nuances of language
codes until I ran into this problem. I was just following the instructions
(or so I thought) on the usage.md file. Maybe other users are more
knowledgeable than I am, but in terms of the script "just working" I think
that would be an enhancement.

**Comment by DavidLRowe** [comment]
@mhosken Suppose I take the data from langtags, look in the Lang_Id column for two-letter language tags (after stripping off any script or region code) then use the corresponding three-letter code from the ISO 639-3 column as the dictionary index for the two-letter language tag. That dictionary could be used in makeoxt to test the language tag, either quitting with an error message, or substituting the two-letter code.

**Comment by DavidLRowe** [comment]
More checking added in v0.4 release (https://github.com/silnrsi/oxttools/releases/tag/v0.4)
